nixos/modules/services/web-servers/unit/default.nix

```diff
@@ -99,17 +99,34 @@ in {
         curl -X PUT --data-binary '@${configFile}' --unix-socket '/run/unit/control.unit.sock' 'http://localhost/config'
       '';
       serviceConfig = {
-        User = cfg.user;
-        Group = cfg.group;
-        AmbientCapabilities = "CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID";
-        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID";
         ExecStart = ''
           ${cfg.package}/bin/unitd --control 'unix:/run/unit/control.unit.sock' --pid '/run/unit/unit.pid' \
                                    --log '${cfg.logDir}/unit.log' --state '${cfg.stateDir}' --no-daemon \
                                    --user ${cfg.user} --group ${cfg.group}
         '';
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Capabilities
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" "CAP_SETGID" "CAP_SETUID" ];
+        # Security
+        NoNewPrivileges = true;
+        # Sanboxing
+        ProtectSystem = "full";
+        ProtectHome = true;
         RuntimeDirectory = "unit";
         RuntimeDirectoryMode = "0750";
+        RuntimeDirectoryPreserve = "yes";
```

Comment:
Why do we set `RuntimeDirectoryPreserve=yes ? Looking at the `preStart` script, the only thing we do in `/run` is remove a file unit might have created during runtime…

Reply:
During debugging, sometimes an error occurs:
```
дек 08 23:03:23 NixOS-Test unitd[8140]: 2019/12/08 23:03:23 [info] 8140#8140 unit started
дек 08 23:03:23 NixOS-Test unitd[8140]: 2019/12/08 23:03:23 [alert] 8140#8140 unlink("/run/unit/unit.pid") failed (2: No such file or directory)
дек 08 23:03:23 NixOS-Test unitd[8140]: ^[[0;1;39m[alert] 8140#8140 unlink("/run/unit/unit.pid") failed (2: No such file or directory)
дек 08 23:03:23 NixOS-Test systemd[1]: ^[[0;1;39m^[[0;1;39munit.service: Main process exited, code=exited, status=1/FAILURE
```
As a result, it is not clear - the process can`t create a folder `/run/unit`, or the problem is different

Reply:
`/run/unit` should be created via the `RuntimeDirectory=unit` directive.

The error is about `unit` trying to remove `/run/unit/unit.pid`. We should disable pid file handling entirely if possible.

Reply:
Removed `RuntimeDirectoryPreserve`.
The unit process automatically creates a pid file, not systemd creates.